### DISCUSSION TITLE
Windows 10 Docker

### DISCUSSION THREAD

**Discussion by mavproductions** [netalertx/NetAlertX#767] [Q&A]
Hi there,
Can someone explain to me why I can't get this working on a windows docker environment? I can't find a working IP for the webUI, and have tried multiple configurations: linux debian on hyper-v instead, WSL2 ubuntu with the docker image. It worked with WSL2 ubuntu with docker image but I couldn't figure out the VLANing to use my usual home LAN network. It stayed within the 172 network I believe and only picked up my router(internet) modem and some other unknown device.

Am I scrub?

Please help! :)

**Comment by jokob-sk** [comment]
Hey @mavproductions , Docker Desktop on Windows doesn't support running containers in network host mode AFAIK

**Reply by jokob-sk** [reply]
If you don't need to rely on-network scans, but use a different discovery method (check the available plugins) you could try to run the container in bridge mode and map the default 20211 port, but I have not tested this approach, so not sure if it works.

**Reply by mavproductions** [reply]
Hey @jokob-sk !

Thanks for the fast replies! Ideally I would like all discovery methods to work (incl on-network scans)

Everytime I use host mode, it appears to be stuck in a 172.xxx network instead of reaching out to my main local network. I can't seem to understand which IP I should ideally be using even though I try accessing all IPs listed under each interface using `ip addr`.

Any tips? Am I over-complicating something here? I just want a simple solution with NetAlertX to use on either my Windows PC or Laptop without impacting performance greatly, and natively accessing any network really I through at it, even though I only need it to access mainly 1 LAN network that I haven't had luck getting it to reach yet.

Using plugins isn't ideal, unless that's normally what people use. I assumed something like SNMP would work alone.

**Reply by jokob-sk** [reply]
As mentioned you can't use for example arp scan on a Windows machine due to docker host limitations. If you use Linux, you can use on network plugin scanners. If so check this guide on subnets: https://github.com/jokob-sk/NetAlertX/blob/main/docs/SUBNETS.md

Every scanner is a Plugin it's how all scanners are implemented. Check this list: 

https://github.com/jokob-sk/NetAlertX/tree/main/front/plugins#readme

You can use the SNMP plugin for discovery as well.

**Reply by mavproductions** [reply]
My mistake for not understanding sorry. So no matter what I really need to use a Linux environment. And WSL2 will cut it, provided I don't use docker?

**Reply by jokob-sk** [reply]
All good! Hummm, I haven't used WSL2 so can't comment. But as mentioned, if you have a router supporting SNMP you can probably use Windows, but you need to pick complementing plugins. Again, not fully tested. The only installation I am capable to fully support due to my other responsibilities is docker on Linux. I'm happy if someone would take up the maintenance of other installation methods 🙂

**Reply by mavproductions** [reply]
So I have host mode working on my Windows Docker Desktop, via **Settings** > **Resources** > **Network**  but it's not really pulling much besides my router and nothing in my main LAN subnet. It pulled from a subnet I have no clue about (192.168.65.xxx) and my router with my main public IP.

I'm assuming in linux, host mode would natively work with a user's main sub network?

**Reply by jokob-sk** [reply]
Hi @mavproductions - You are in uncharted territory so I will be of no use here. I would assume there is an interface and subnet you would specify depending on the host machine. Did you check what interfaces / networks are available under About -> System info ->  Network Hardware? I would try the interfaces / masks listed there : 

![image](https://github.com/user-attachments/assets/a85c92cd-5b5b-4e4c-9de7-2ca838e259a1)

**Reply by mavproductions** [reply]
I setup a ubuntu VM and wiped my windows netalertx setup and it didn't work immediately. I think I just had to setup the following as interfaces on my windows docker container - but I haven't tested it yet.
`192.168.1.0/24 --interface=eth1`
`192.168.2.0/24 --interface=ens33`
`192.168.2.0/24 --interface=eth1`

one of the last 2 on my ubuntu VM made my real network appear, so I just need to try plugging away with similar on my docker container. for now I'm keeping with the ubuntu VM as I had a lot of networking issues happening with windows docker. 

Ideally I get the windows docker desktop setup working as it's less resources, but frankly my ubuntu vm is less of a headache right now and I can snapshot everything as working really nicely :)

**Reply by jokob-sk** [reply]
Glad to hear you have a working solution! If you ever figure out how to get a Windows setup working, please let me know or, even better, write up a guide and I will include it in the docs. 🙂 Looking forward if you decide to contribute to the project!
